coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -254,31 +256,128 @@ def extract_documentation_with_markers(content, docstyle_definition):
-    :raises FileNotFoundError: Raised when the docstyle definition file was not
-                               found.
-    :raises KeyError:          Raised when the given language is not defined in
-                               given docstyle.
-    :raises ValueError:        Raised when a docstyle definition setting has an
-                               invalid format.
-    :return:                   An iterator returning each DocumentationComment
-                               found in the content.
+    DocBaseClass holds important function which will extract, fix parse data
+    and generates diffs for documentation.
     """
-    docstyle_definition = DocstyleDefinition.load(language, docstyle)
-    return extract_documentation_with_markers(content, docstyle_definition)
+
+    def __init__(self, filename, file):
+        """
+        Instantiates a new DocBaseClass
+
+        :param file:
+            contains file content
+        :param filename:
+            contains file name
+        """
+        self.warning_desc = None
+        self.affected_code = None
+        self.diffs = None
+        self.file = file
+        self.filename = filename
+
+    def extraction(self, content, language, docstyle):
+        """
+        Extracts all documentation texts inside the given source-code-string
+        using the coala docstyle definition files.
+
+        The documentation texts are sorted by their order appearing in
+        ``content``.
+
+        For more information about how documentation comments are
+        identified and extracted, see DocstyleDefinition.doctypes enumeration.
+
+        :param content:            The source-code-string where to extract
+                                   documentation from. Needs to be a list or
+                                   tuple where each string item is a single
+                                   line (including ending whitespaces like
+                                   ``\\n``).
+        :param language:           The programming language used.
+        :param docstyle:           The documentation style/tool used
+                                   (e.g. doxygen).
+        :raises FileNotFoundError: Raised when the docstyle definition file
+                                   was not found.
+        :raises KeyError:          Raised when the given language is not
+                                   defined in given docstyle.
+        :raises ValueError:        Raised when a docstyle definition setting
+                                   has an invalid format.
+        :return:                   An iterator returning each
+                                   DocumentationComment found in the content.
+        """
+        docstyle_definition = DocstyleDefinition.load(language, docstyle)
+        return extract_documentation_with_markers(
+                        content, docstyle_definition)
+
+    def fix_parse_data(self, parsed, allow_missing_func_desc: str=False,
```

Comment:
Also, why is the fixing part of the base class? Should that not be what the Bear does?